odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1700,6 +1700,52 @@ def assertBinaryValue(record, value):
         self.assertEqual(record_no_bin_size.binary_computed, expected_value)
         self.assertEqual(record_bin_size.binary_computed, expected_value)
 
+    def test_96_ir_rule_m2m_field(self):
+        """Ensures m2m fields can't be read if the left records can't be read.
+        Also makes sure reading m2m doesn't take more queries than necessary."""
+        tag = self.env['test_new_api.multi.tag'].create({})
+        record = self.env['test_new_api.multi.line'].create({
+            'name': 'image',
+            'tags': [(4, tag.id)],
+        })
+
+        # only one query as admin: reading pivot table
+        with self.assertQueryCount(1):
+            record.read(['tags'])
+
+        record_user = record.with_user(self.env.ref('base.user_demo'))
+
+        # prep the following query count by caching access check related data
+        record_user.read(['tags'])
+
+        # only one query as user: reading pivot table
+        with self.assertQueryCount(1):
+            record_user.read(['tags'])
+
+        # create a passing ir.rule
+        self.env['ir.rule'].create({
+            'model_id': self.env['ir.model'].search([('model', '=', record._name)]).id,
```

Comment:
```suggestion
            'model_id': self.env['ir.model']._get(record._name).id,
```

Reply:
I think I copied that so FYI they are other occurrences of this in our code, but OK to change this new one.